### ISSUE TITLE
Waypoints in SharePoint

### ISSUE THREAD

**Issue by wpsmith** [imakewebthings/waypoints#374]
Sadly, SharePoint screws up `pageYOffset` and `pageXOffset` leaving it as 0 (the only one I care about it `pageYOffset`). So `this.adapter.scrollTop()` always equals 0. In both jquery.waypoints.js and waypoints.js, this is faults and errors out. In jQuery, `jQuery('selector').offset().top` works for the top.

I would really love help in fixing this in a couple ways:

1. Add an event to hook into, trigger, and/or over-ride the waypoint check. There is probably way to do this as I haven't had the time to do a deep dive yet. In SharePoint, since the `pageYOffset` and `pageXOffset` always remain 0, the `scroll.waypoints` and `resize.waypoints` never trigger.
1. Create a hack patch to get me through this hiccup that works either with the noframework or jQuery scripts.

Please feel free to reach out to me for more information and/or an environment to have this reproduced and testing.

**Comment by imakewebthings** [comment]
I have no experience with SharePoint, so the following suggestions are out of thin air. If you would like to share with me a link to the actual site produced, you have my email.

- Is the window/body/html the element being scrolled? If, perhaps, the scrolling element isn't the window but some other scrollable element consider setting the [context option](http://imakewebthings.com/waypoints/api/context-option/).
- The way [adapters work](http://imakewebthings.com/waypoints/guides/adapters/) you can redefine what "scrollTop" ends up running on the element. You could either fork one of the existing adapters and modify it for whatever SharePoint's proprietary offset needs are, or use an existing adapter and override the adapter definitions, e.g. `Waypoint.Adapter.prototype.scrollTop = function() { ... }`

**Comment by wpsmith** [comment]
Thanks for the help. I had to create a custom Adapter, only over-riding 2 prototype functions, and re-assign the context option. I'll add this to my fork soon.

**Comment by Suresh-Lakum** [comment]
I am facing similar issue can you provide me hack you have applied @wpsmith ?

**Comment by wpsmithtwc** [comment]
@Suresh-Lakum I do not have access to the code, and I failed to upload a PR or a demonstration. However, if memory serves me correctly, it has something to do with how SharePoint handles `window.pageXOffset` and `window.pageYOffset`. I will get my old computer and see if I can dig it back up, but I doubt I can.

**Comment by wpsmith** [comment]
@Suresh-Lakum Check out this gist: https://gist.github.com/wpsmithtwc/d448f7edaed4646b7070d956abf720fb

It contains the code I ran in production. I dumped all files that I thought my be relevant. I have the entire sln folder for Visual Studio but I left that world in 2015 and have frankly forgotten much of it. But if you are good with JS and know SharePoint JS best practices (e.g., wrap all functions in a SP object, use `_EnsureJSNamespace` & `SP.SOD`, etc), then you will be fine.
